### ISSUE TITLE
Throwing MultipleFailureException with empty list of throwables forces test to pass 

### ISSUE THREAD

**Issue by idubrov** [junit-team/junit4#1169]
Throwing "new MultipleFailureException(Collections.emptyList())" makes current test to succeed.

I think this behavior is confusing and even dangerous. If you are throwing MultipleFailureException you are explicitly indicating test should fail. However, if for some reason (let's say bugs in your test code) you fail to correctly assemble list of throwables, throwing MultipleFailureException would actually make test to bypass all the remaining assertions and succeed!

For example, the following test would succeed no matter that assertEquals should trigger test failure:

``` java
@Test
public void testSucceeds() throws Exception {
  makeTestSucceed();
  assertEquals(1, 2);
}
private void makeTestSucceed() throws Exception {
  throw new MultipleFailureException(Collections.emptyList());
}
```

**Comment by kcooney** [comment]
I think it would be reasonable for the `MultipleFailureException` constructor to throw an `IllegalArgumentException` if you pass in an empty list.

@marcphilipp do you agree? If so, I think we should do this in the JUnit5 branch, since it could cause some passing tests to fail.

**Comment by marcphilipp** [comment]
+1

**Comment by kcooney** [comment]
Thanks, @idubrov for reporting the issue. Would you be willing to send us a fix and an associated test?

**Comment by marcphilipp** [comment]
Fixed by 31f72d37e12a134de67d89bf5504df43e0858867.

**Comment by marcphilipp** [comment]
This issue was fixed in the `junit5` branch. However, that branch will be closed now. Therefore, I'm re-opening this issue.

**Comment by jbduncan** [comment]
@kcooney, what else needs to be done to fulfill this issue, if anything?

**Comment by kcooney** [comment]
@jbduncan as Marc said, this was fixed in the junit5 branch but for various reasons we cannot merge that branch to master. If anyone wants this fixed they can send us a fresh pull from a branch off of master.

**Comment by panchenko** [comment]
This can be closed as it is fixed by #1376

**Comment by marcphilipp** [comment]
Thanks, @panchenko!
